test/functional/ui/input_spec.lua

```diff
@@ -406,6 +406,61 @@ describe('input non-printable chars', function()
   end)
 end)
 
+describe('skip do you really want to write on force write', function()
+  after_each(function()
+    os.remove('Xtest-overwrite-forced')
+  end)
+
+  it("does not ask for confirmation when force write", function()
+    write_file("Xtest-overwrite-forced", [[foobar]])
+    local screen = Screen.new(40,4)
+    screen:set_default_attr_ids({
+      [1] = {bold = true, foreground = Screen.colors.Blue1},
+      [2] = {foreground = Screen.colors.Grey100, background = Screen.colors.Red},
+      [3] = {bold = true, foreground = Screen.colors.SeaGreen4}
+    })
+    screen:attach()
+    command("set display-=msgsep shortmess-=F")
+
+    feed_command("e Xtest-overwrite-forced")
+    screen:expect([[
+      ^foobar                                  |
+      {1:~                                       }|
+      {1:~                                       }|
+      "Xtest-overwrite-forced" [noeol] 1L, 6B |
+    ]])
+
+    -- The timestamp is in second resolution, wait two seconds to be sure.
+    screen:sleep(2000)
+    write_file("Xtest-overwrite-forced", [[smurf]])
+    feed_command("w")
+    screen:expect([[
+      {2:WARNING: The file has been changed since}|
+      {2: reading it!!!}                          |
+      {3:Do you really want to write to it (y/n)^?}|
+                                              |
+    ]])
+
+    feed("n")
+    feed("<cr>")
+    screen:expect([[
+      ^foobar                                  |
+      {1:~                                       }|
+      {1:~                                       }|
+                                              |
+    ]])
+
+    feed("w!")
+    screen:expect([[
+      fooba^r                                  |
+      {1:~                                       }|
+      {1:~                                       }|
+                                              |
+    ]])
+    feed("n")
```

Comment:
would use `screen:expect_unchanged()` here. But really it's not necessary.
Instead use command() because it's synchronous (whereas feed() is async), then
check v:errmsg.

```
command('let v:errmsg = ""')
command('w!')
eq('', eval('v:errmsg'))
```

Reply:
I accidentally called feed() instead of feed_command(). When calling the proper method, the screens differ.

Reply:
if possible, use `command()` (synchronous and error-checked) instead `feed_command()` (asyncrhonous, errors are not surfaced in the test harness)

Reply:
I made the changes, but I had to use `feed_command("w")` when testing that the prompt appears otherwise the test would just hang, I added a comment.